src/caja-desktop-window.c

```diff
@@ -204,6 +274,8 @@ caja_desktop_window_new (CajaApplication *application,
         XFree(xch);
     }
 
+    gtk_widget_hide (CAJA_WINDOW (window)->details->menubar);
+
```

Comment:
Is this still needed?

Reply:
Looks like it is not: we have
```
    gtk_widget_hide (CAJA_WINDOW (window)->details->statusbar);
    gtk_widget_hide (CAJA_WINDOW (window)->details->menubar);
```
in caja_desktop_window_init and that works unless `gtk_widget_show_all` is called somewhere. The x11 code uses `gtk_widget_realize()` and due to a problem with that in wayland we use g`tk_widget_show` as using `gtk_widget_realize()` in wayland with and only with gtk-layer-shell gets us this error and blocks showing the desktop:
`
(caja:30733): Gdk-WARNING **: 13:52:09.061: (../gdk/wayland/gdkwindow-wayland.c:915):buffer_release_callback: runtime check failed: (impl->staging_cairo_surface != cairo_surface)`

With gtk_widget_show we avoid that, but if we use  gtk_widget_show_all() we have to re-hide the menubar and statusbar. 

Thus it seems safe to remove the second instance of 
`gtk_widget_hide (CAJA_WINDOW (window)->details->menubar);`
from the x11 code as well. Can put it back if others want me to so as to avoid any change in the x11 code

Reply:
What do you mean?  It wasn't in the original X11 code path, was it?

The reason why `show_all()` caused a problem is that what it does is recursively call `show()` on itself and all public children, which we don't want here.